core/src/main/kotlin/discord4j/core/ExtensionExtensions.kt

```diff
@@ -0,0 +1,18 @@
+package discord4j.core
```

Comment:
I really do not understand the point of this class in large. I feel anything that would be useful should be public, not internal. `unit` and Optional to nullable type transformations are probably the most useful, but everything else in here seems rather pointless to me. `Mono#awaitNull`, `Mono#await`, `Flux#infinite` especially are very useless to me.